### DISCUSSION TITLE
Hit `C-c C-n` multiple times

### DISCUSSION THREAD

**Discussion by aadcg** [manna-harbour/miryoku#237] [Q&A] [answered]
In Emacs, the keybinding is frequently used several times in a row. On a typical keyboard, the pinky can hold the Control key while `n` and `p` can be pressed very quickly and as many times as desired without changing position significantly. 

Compare it to Miryoku. On a QWERTY keyboard: right hand holds Control as a home mod, left hands hits `c`; left hand holds Control as a home mod, right hands hits `n`. Repeat. This is not very efficient.

I wonder how people cope with this inefficiency. Thanks.

**Comment by manna-harbour** [comment, marked as answer]
For that particular situation I use the Nav layer.

Outside of that there's still a potential issue with alternating Control but it hasn't been a problem for me in practice. 

https://reddit.com/r/emacs/s/HpLtXndMB1
> I'm lucky in that the long control sequence I use most C-c C-v (C-)t (org-babel-tangle) is all one one hand, and other frequent long sequences like rectangle editing only need the mod at the beginning (e.g. C-x r k (kill-rectangle)).
>
> Control on home row mods is a similar issue to shift. It's annoying at first but if you practice you'll build the muscle memory of which hand to use for mods for each letter, then alternating won't be such an issue.
>
> For shift, for single letters, auto shift (or retro shift for home row mods) is quite nice. For more than 3 consecutive capitals I'll use caps lock. At some point I'll add capsword to Miryoku, which is caps lock with auto disable. Shift on thumb, particularly one shot shift, is also worth considering, but that requires a spare thumb key.
>
>For the control issue there are analogous approaches.  Most of the awkwardness will go away with practice, as with shift. You could use control lock for long sequences, using the os sticky keys feature. Or you could use tap-hold to send the control sequence on hold (although the home row would require a modified retro shift for that). You could also use control on a thumb. But for emacs, even without home row mods, the best way is probably to use something like transient or hydra, where you only need the mod at the beginning of the sequence.

**Comment by aadcg** [comment]
> For that particular situation I use the Nav layer.

I'm not sure if I follow. I was referring to commands `comint-previous-prompt` and `comint-next-prompt` (bound to `C-c C-p` and `C-c C-p` respectively). I bounded them to `M-<up>` and `M-<down>` so that I can hold the modifier key and hit the arrow keys at will. For this particular case, I think it's the best solution.

> Or you could use tap-hold to send the control sequence on hold (although the home row would require a modified retro shift for that).

Could you explain in more detail please? I think that a way to solve this issue would be to make use of `QK_LOCK` (though I have never tried).

> You could also use control on a thumb.

This is what I have been doing all along. I use home row mods but I have "thumb home row mods" too (`C` under my left thumb, `M` under my right thumb on hold). For anyone interested,[ see my config](https://github.com/aadcg/qmk_firmware/blob/sofle/keyboards/aadcg_sofle/keymaps/default/keymap.c).
